src/olympia/accounts/tests/test_utils.py

```diff
@@ -247,20 +268,39 @@ def test_success(self, primary_email_change_event):
         process_fxa_event(self.body)
         primary_email_change_event.assert_called()
         primary_email_change_event.assert_called_with(
-            'new-email@example.com', 'ABCDEF012345689',
+            'ABCDEF012345689',
+            totimestamp(self.email_changed_date),
+            'new-email@example.com')
+
+
+class TestProcessFxAEventDelete(TestCase):
+    def setUp(self):
+        self.email_changed_date = self.days_ago(42)
+        self.body = json.dumps({'Message': json.dumps(
+            {'event': 'delete',
+             'uid': 'ABCDEF012345689',
+             'ts': totimestamp(self.email_changed_date)})})
+
+    @override_switch('fxa-account-delete', active=True)
+    def test_success_integration(self):
+        user = user_factory(fxa_id='ABCDEF012345689')
+        process_fxa_event(self.body)
+        user.reload()
+        assert user.email is None
+        assert user.deleted
+        assert user.fxa_id is None
+
+    @override_switch('fxa-account-delete', active=True)
+    @mock.patch('olympia.accounts.utils.delete_user_event.delay')
+    def test_success(self, delete_user_event_mock):
```

Comment:
Do we really need this if we also have the integration test, above? I guess this ensures that the results are due to `delete_user_event` being called, and not some other side effect? Is there a way to combine the two and spy on `delete_user_event` but also have it actually execute? We can do that in JS tests but I'm not sure if `unittest` supports that.

Reply:
it's possible, but a pain to do with mock, so we generally don't do it.  (Normally we would just have either the integration test _or_ the mocked test - I must have been feeling extra cautious when I added the tests for the earlier integration, and I just copied them here)